sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -36,6 +36,16 @@
 from sklearn.utils.testing import assert_warns
 
 
+def test_deprecation_of_n_components_in_linkage_tree():
+    rng = np.random.RandomState(0)
+    X = rng.randn(50, 100)
+    # Test for warning of deprecation of n_components in linkage_tree
+    children, n_nodes, n_leaves, parent = assert_warns(
+                                        DeprecationWarning,
+                                        linkage_tree,
+                                        X.T,
+                                        n_components=10)
+
```

Comment:
I had suggested comparing `children, n_nodes, n_leaves, parent` to `linkage_tree(X.T)`